zap/src/main/java/org/zaproxy/zap/core/scanner/InputVectorBuilder.java

```diff
@@ -0,0 +1,109 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.core.scanner;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.parosproxy.paros.core.scanner.NameValuePair;
+import org.zaproxy.zap.core.scanner.InputVector.PayloadFormat;
+
+/**
+ * {@code InputVectorBuilder} provides an easy way to build {@code InputVectors} and validates the
+ * provided inputs.
+ *
+ * @author preetkaran20@gmail.com KSASAN
+ * @since 2.11.0
+ */
+public class InputVectorBuilder {
+
+    private Map<Integer, InputVector> inputVectorsMap = new HashMap<>();
+
+    /**
+     * Note: calling this method with same NameValuePair will overwrite the earlier name and value
+     * with the newer one.
+     *
+     * @param nameValuePair
+     * @param param
+     * @param value
+     * @param namePayloadFormat
+     * @param valuePayloadFormat
+     * @return {@code InputVectorBuilder}
+     */
+    public InputVectorBuilder setNameAndValue(
+            NameValuePair nameValuePair,
+            String param,
+            String value,
+            PayloadFormat namePayloadFormat,
+            PayloadFormat valuePayloadFormat) {
+        inputVectorsMap.put(
+                nameValuePair.getPosition(),
+                new InputVector(
+                        nameValuePair.getPosition(),
+                        param,
+                        value,
+                        namePayloadFormat,
+                        valuePayloadFormat));
+        return this;
+    }
+
+    /**
+     * Note: calling this method with same NameValuePair will overwrite the earlier name and value
+     * with the newer one.
+     *
+     * @param nameValuePair
+     * @param value
+     * @param payloadFormat
+     * @return {@code InputVectorBuilder}
+     */
+    public InputVectorBuilder setValue(
+            NameValuePair nameValuePair, String value, PayloadFormat payloadFormat) {
+        return this.setNameAndValue(
+                nameValuePair,
+                nameValuePair.getName(),
+                value,
+                PayloadFormat.ALREADY_ESCAPED,
+                payloadFormat);
+    }
+
+    /**
+     * Note: calling this method with same NameValuePair will overwrite the earlier name and value
+     * with the newer one.
+     *
+     * @param nameValuePair
+     * @param param
+     * @param payloadFormat
+     * @return {@code InputVectorBuilder}
+     */
+    public InputVectorBuilder setName(
+            NameValuePair nameValuePair, String param, PayloadFormat payloadFormat) {
+        return this.setNameAndValue(
+                nameValuePair,
+                param,
+                nameValuePair.getValue(),
+                payloadFormat,
+                PayloadFormat.ALREADY_ESCAPED);
+    }
+
+    public List<InputVector> build() {
+        return new ArrayList<>(inputVectorsMap.values());
```

Comment:
Better if we return a consistent order, most of the variants would benefit of that, e.g.:
```Java
return inputVectorsMap.values().stream().sorted(Comparator.comparingInt(InputVector::getPosition)).collect(Collectors.toList());
```

Reply:
as we are using HashMap with key position, hence it becomes somehow consistent order but will do.

Reply:
done

Reply:
`HashMap` is not a `SortedMap` though.

Reply:
yeah, may be if size is more than 12, then i think hashcode will not give increasing order.